pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -944,3 +885,87 @@ func isUpdating(ctx *zedagentContext) bool {
 	}
 	return false
 }
+
+//schedulePublishMessageToZedCloud put message into queue
+func schedulePublishMessageToZedCloud(ctx *zedagentContext, ReportInfo *info.ZInfoMsg, uuid string) {
+	data, err := proto.Marshal(ReportInfo)
+	if err != nil {
+		log.Fatal("sendInfoMessageToZedCloud proto marshaling error: ", err)
+	}
+	ctx.infoQueue.Put(ReportInfo.GetZtype(), uuid, data, true)
+}
+
+//infoProcessingTask process infoQueue and send scheduled messages to the controller
+func infoProcessingTask(ctx *zedagentContext) {
+	// in case of 10 errors in row sleep before next send
+	errorsWatermark := 10
+	// sleep before next send duration in case of errors
+	sleepOnErrors := 10 * time.Second
+	// Publish info every 90-300 Millisecond if scheduled
```

Comment:
Why do we want to slow this down? Seems like we should trigger the sending of info messages and then send them all.